### PR TITLE
Make indexer listen to delete record events and trim those records from the index

### PR DESCRIPTION
**Pull request by AlexGilleran**
### What this PR does
- Indexer now registers its webhook to include events and listen to DeleteRecord events
- Indexer now checks the events in webhook payloads for DeleteRecord events and deletes them from the index
- Indexer now updates its webhook on every start-up, even if already registered
- Minikube helm config default has registry authentication off

### Checklist
- [x] There are unit tests to verify my changes are correct | Unit tests aren't applicable (delete one)
- [x] I've updated CHANGES.md with what I changed.
- [x] I've linked this PR to an issue in ZenHub (or there is no issue) and dragged it to the _Pull Request_ column

### CODE REVIEW

File: magda-indexer/src/main/scala/au/csiro/data61/magda/indexer/external/registry/RegisterWebhook.scala [hunk 1]
```diff
@@ -58,18 +61,20 @@ object RegisterWebhook {
         EventType.DeleteRecord,
         EventType.CreateRecordAspect,
         EventType.PatchRecord,
-        EventType.DeleteRecordAspect),
+        EventType.DeleteRecordAspect,
+        EventType.DeleteRecord),
```
Comment (kring) [inline magda-indexer/src/main/scala/au/csiro/data61/magda/indexer/external/registry/RegisterWebhook.scala]: `DeleteRecord` already exists, third from the top.

File: magda-indexer/src/main/scala/au/csiro/data61/magda/indexer/external/registry/WebhookApi.scala [hunk 2]
```diff
@@ -32,23 +34,38 @@ class WebhookApi(indexer: SearchIndexer)(implicit system: ActorSystem, config: C
     magdaRoute {
       post {
         entity(as[WebHookPayload]) { payload =>
-          payload.records match {
-            // TODO: Handle registry config not found
-            case Some(records) =>
-              val dataSets = records.map(record => try {
+          val events = payload.events match {
```
Comment (kring) [inline magda-indexer/src/main/scala/au/csiro/data61/magda/indexer/external/registry/WebhookApi.scala]: `val events = payload.events.getOrElse(List())`?

File: magda-indexer/src/main/scala/au/csiro/data61/magda/indexer/external/registry/WebhookApi.scala [hunk 3]
```diff
@@ -32,23 +34,38 @@ class WebhookApi(indexer: SearchIndexer)(implicit system: ActorSystem, config: C
...
+            case None       => List()
+            case Some(list) => list
+          }
+
+          val idsToDelete = events.filter(_.eventType == EventType.DeleteRecord)
```
Comment (kring) [inline magda-indexer/src/main/scala/au/csiro/data61/magda/indexer/external/registry/WebhookApi.scala]: What if the record is deleted and quickly recreated (i.e. by events in the same webhook payload)?

File: magda-indexer/src/main/scala/au/csiro/data61/magda/indexer/external/registry/RegisterWebhook.scala (hunk 1 above)
Comment (AlexGilleran) [reply magda-indexer/src/main/scala/au/csiro/data61/magda/indexer/external/registry/RegisterWebhook.scala]: Good point...

File: magda-indexer/src/main/scala/au/csiro/data61/magda/indexer/external/registry/WebhookApi.scala (hunk 2 above)
Comment (AlexGilleran) [reply magda-indexer/src/main/scala/au/csiro/data61/magda/indexer/external/registry/WebhookApi.scala]: I get paid per LOC right? Fixed 😳

File: magda-indexer/src/main/scala/au/csiro/data61/magda/indexer/external/registry/WebhookApi.scala (hunk 3 above)
Comment (AlexGilleran) [reply magda-indexer/src/main/scala/au/csiro/data61/magda/indexer/external/registry/WebhookApi.scala]: Good point. Wrote a test for this and somehow it works but I think that's just luck - have made the execution order explicit (delete everything then create everything).

**Comment by AlexGilleran** [comment]
@kring addressed the review.

**Pull request merged**